pkg/gofr/service/openai/openai.go

```diff
@@ -0,0 +1,190 @@
+package openai
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"go.opentelemetry.io/otel/trace"
+)
+
+type Config struct {
+	APIKey       string
+	Model        string
+	BaseURL      string
+	Timeout      time.Duration
+	MaxIdleConns int
+}
+
+type Client struct {
+	config     *Config
+	logger     Logger
+	metrics    Metrics
+	tracer     trace.Tracer
+	httpClient *http.Client
+}
+
+var (
+	errorMissingAPIKey = errors.New("api key not provided")
```

Comment:
```suggestion
	errorMissingAPIKey = errors.New("API key not provided")
```

API is an acronym: Advanced Processing Interface